test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs

```diff
@@ -91,6 +101,201 @@ public async Task TestRecoveryAfterFailedExport()
         await host.StopAsync();
     }
 
+    // For `Grpc.Core.StatusCode.DeadlineExceeded`
+    // See https://github.com/open-telemetry/opentelemetry-dotnet/issues/5436.
+    [Theory]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.Unavailable)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.Cancelled)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.Aborted)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.OutOfRange)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.DataLoss)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.Internal)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.InvalidArgument)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.Unimplemented)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.FailedPrecondition)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.PermissionDenied)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.Unauthenticated)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.DeadlineExceeded)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.Unavailable)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.Cancelled)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.Aborted)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.OutOfRange)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.DataLoss)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.Internal)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.InvalidArgument)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.FailedPrecondition)]
+    [InlineData(false, ExportResult.Failure, Grpc.Core.StatusCode.DeadlineExceeded)]
+    public async Task GrpcRetryTests(bool useRetryTransmissionHandler, ExportResult expectedResult, Grpc.Core.StatusCode initialStatusCode)
+    {
+        var testGrpcPort = Interlocked.Increment(ref gRPCPort);
+        var testHttpPort = Interlocked.Increment(ref httpPort);
+
+        using var host = await new HostBuilder()
+           .ConfigureWebHostDefaults(webBuilder => webBuilder
+                .ConfigureKestrel(options =>
```

Comment:
nit: We could speed up the tests by clearing the default logging providers that get added when building the hosts.

```csharp
.ConfigureLogging(loggingBuilder => loggingBuilder.ClearProviders())
```